### PR TITLE
Fix some typo and unify the AOT file format for init data

### PR DESCRIPTION
**Pull request by TianlongLiang**
Unify the emitting and loading logic for table/memory/global init data

### CODE REVIEW

File: core/iwasm/aot/aot_loader.c [hunk 1]
```diff
@@ -1169,17 +1176,30 @@ load_init_expr(const uint8 **p_buf, const uint8 *buf_end, AOTModule *module,
...
-            read_byte_array(buf, buf_end, &expr->u.v128, sizeof(expr->u.v128));
+            i64x2 = (uint64 *)expr->u.v128.i64x2;
+            CHECK_BUF(buf, buf_end, sizeof(uint64) * 2);
+            wasm_runtime_read_v128(buf, &i64x2[0], &i64x2[1]);
+            buf += sizeof(uint64) * 2;
             break;
         case INIT_EXPR_TYPE_GET_GLOBAL:
             read_uint32(buf, buf_end, expr->u.global_index);
             break;
+#if WASM_ENABLE_GC == 0
```
Comment (wenyongh) [inline core/iwasm/aot/aot_loader.c:1187]: Had better be `#if WASM_ENABLE_GC == 0 && WASM_ENABLE_REF_TYPES != 0` like other places.

File: core/iwasm/aot/aot_loader.c [hunk 2]
```diff
@@ -1169,17 +1176,30 @@ load_init_expr(const uint8 **p_buf, const uint8 *buf_end, AOTModule *module,
...
+        case INIT_EXPR_TYPE_FUNCREF_CONST:
         case INIT_EXPR_TYPE_REFNULL_CONST:
-            read_uint32(buf, buf_end, expr->u.type_index);
+#if UINTPTR_MAX == UINT64_MAX
+            read_uint64(buf, buf_end, expr->u.ref_index);
+#else
+            read_uint32(buf, buf_end, expr->u.ref_index);
```
Comment (wenyongh) [inline core/iwasm/aot/aot_loader.c:1193]: Found that `void *data;` field of WASMValue is added in `core/iwasm/gc_export.h`, it isn't added in WASMValue definition in `core/iwasm/interpreter/wasm.h`, could you help add it?

File: core/iwasm/aot/aot_loader.c [hunk 3]
```diff
@@ -2119,32 +2130,14 @@ load_globals(const uint8 **p_buf, const uint8 *buf_end, AOTModule *module,
...
     /* Create each global */
     for (i = 0; i < module->global_count; i++) {
-#if WASM_ENABLE_GC == 0
-        uint16 init_expr_type;
-#endif
-
         read_uint8(buf, buf_end, globals[i].type);
         read_uint8(buf, buf_end, globals[i].is_mutable);
+        /* Skip 2 byte for padding */
+        buf += 2;
```
Comment (wenyongh) [inline core/iwasm/aot/aot_loader.c:2136]: Is it better to use `buf = align_ptr(buf, 4);`, and ensure the same alignment when emitting AOT file?

File: core/iwasm/compilation/aot_emit_aot_file.c [hunk 4]
```diff
@@ -291,8 +297,21 @@ get_init_expr_size(const AOTCompContext *comp_ctx, const AOTCompData *comp_data,
...
             break;
+        case INIT_EXPR_TYPE_FUNCREF_CONST:
         case INIT_EXPR_TYPE_REFNULL_CONST:
-            /* type_index */
+            if (!comp_ctx->enable_gc) {
+                /* ref_index */
+                size += comp_ctx->pointer_size;
+                break;
+            }
+#if WASM_ENABLE_GC != 0
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_emit_aot_file.c:307]: Not a good coding style, had better:
```C
        case INIT_EXPR_TYPE_FUNCREF_CONST:
        case INIT_EXPR_TYPE_REFNULL_CONST:
            if (!comp_ctx->enable_gc) {
                ...
            }
            else {
                 ...
            }
            break;
#if WASM_ENABLE_GC != 0
       case INIT_EXPR_TYPE_I31_NEW:
       ...
#endif
```

File: core/iwasm/compilation/aot_emit_aot_file.c [hunk 5]
```diff
@@ -1831,12 +1819,22 @@ aot_emit_init_expr(uint8 *buf, uint8 *buf_end, uint32 *p_offset,
...
+            if (!comp_ctx->enable_gc) {
+                if (comp_ctx->pointer_size == 4) {
+                    EMIT_U32(expr->u.ref_index);
+                }
+                else {
+                    EMIT_U64(expr->u.ref_index);
+                }
+                break;
+            }
+#if WASM_ENABLE_GC != 0
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_emit_aot_file.c:1833]: Same as above

File: core/iwasm/compilation/aot_emit_aot_file.c [hunk 6]
```diff
@@ -2244,21 +2225,11 @@ aot_emit_global_info(uint8 *buf, uint8 *buf_end, uint32 *p_offset,
...
-#endif /* end of WASM_ENABLE_GC != 0 */
-        {
-            EMIT_U16(global->init_expr.init_expr_type);
-            if (global->init_expr.init_expr_type != INIT_EXPR_TYPE_V128_CONST)
-                EMIT_U64(global->init_expr.u.i64);
-            else
-                EMIT_V128(global->init_expr.u.v128);
-        }
+        /* padding */
+        EMIT_U16(0);
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_emit_aot_file.c:2229]: How about `offset = align_uint(offset, 4)`

**Review by wenyongh — APPROVED** [review]
LGTM

**Review by xujuntwt95329 — COMMENTED** [review]
LGTM

**Pull request merged**